### ISSUE TITLE
Limit plots tab to 1500 items

### ISSUE THREAD

**Issue by aislinnpearson** [Rothamsted-Ecoinformatics/farm_rothamsted#362]
Currently we have an issue where if a treatment has a lot of plots associated with it you need to select 'all plots' to be able to get the plots to load in a quick form. This is prone to error, and we woudl prefer it if all treatments appeared as default.

Displaying 'all plots' when there is a large number of plots also seems to result in a visualisation error. For example this search query requests several hundred plots are displayer: 

![image](https://user-images.githubusercontent.com/16349781/220092242-0ec50e48-b01d-4c70-aeef-8f58ad0acdde.png)

Producing the following error:

![image](https://user-images.githubusercontent.com/16349781/220092504-04aab844-9c24-4b1d-b5ff-a380e820b244.png)

We may want to create a new issue for the error, but put together as a single related issue for now.

**Comment by paul121** [comment]
> Displaying 'all plots' when there is a large number of plots also seems to result in a visualisation error. For example this search query requests several hundred plots are displayer:

Oops @aislinnpearson  we didn't discuss that you sometimes see this error when displaying all of the plots on some plans. Because of this I'm more hesitant to default to showing all of the plots, because it would directly show the error, and never give users the chance to see *any* plots...

That said... testing this now in both my local site and production, I don't seem to be getting this error anymore 🤔 This issue was opened some time ago so perhaps something has changed?

I was going to ask you @mstenta if simply increasing the amount of memory available to PHP on their production server might be possible? I'm not sure what the specs are but if we can reasonably increase this, then it might be worth looking into. Assuming that this error might be related to that...

**Comment by mstenta** [comment]
Sure - I temporarily increased the `memory_limit` to `512M`. Let me know if that helped, and I can make it permanent.

**Comment by aislinnpearson** [comment]
Thanks both. Does this work @paul121 or do you want me to re-specify the issue?

I have checked a couple of our biggest experiments and all the plots do load:

https://rothamstedfarm.farmos.net/plan/102/plots?name=&status=active&items_per_page=All

![Image](https://github.com/user-attachments/assets/524ac33f-607c-46c0-9b1d-90ac8465aca1)

**Comment by paul121** [comment]
Great! Glad that is working. Let's move forward with this 👍

**Comment by aislinnpearson** [comment]
Hey @paul121,

So I had a g with this experiment on the test site and had what I assume is a time-out error related to the number of plots:

https://pr-19-j7qtqgy-2mu45anerqtvq.uk-1.platformsh.site/plan/42

![Image](https://github.com/user-attachments/assets/adbd4c7d-cc46-4eca-bb7d-406fb3fcb86d)

I wonder if we need to rethink this and set some kind of limit (perhaps 500 or 1,000 plots - I think our internal highest is 1bout 2,000) with a warning message or something if additional plots are in a second window?

One to chat about this evening.

**Comment by paul121** [comment]
Drat. I'm seeing this on the production site as well. Hmm. I agree, displaying a warning might be the best option here.

**Comment by aislinnpearson** [comment]
Hi @paul121,

So I was talking to our trials manager and he had a suggestion for this. He said the maximum number of plots we have in a trial at the moment is 3,000 but these are almost always variety trials, meaning you'd very rarely filter them by variety as this is usually added as a single drilling record to the entire boundary (on some occasions you might only drill half an experiment - for example if the plot drill fails half way through - but they always aim to drill everything on the same day).

On the rare occasions that these trials do have a second treatment, there are usually 2 -4 replicates of each treatment so you'd only need to select between half and a quarter of the plots.

So, when it comes to QA (which is the primary purpose of this issue - reducing user error, especially by those unaware you need to press 'view all') the maximum number of plots to display for a trial with 3,000 plots is between 750 and 1,500. For a first release, and to get this out the door, a page display limit of 1,500 by default would be ideal - but I would settle for as close a number to this as we can get depending on processing memory. I can then add a follow up issue for the warning message - which I think should definitely be there, but then won't be holding up the development of this issue. 

Jonah also asked if we could add a filter to the plots for Plot ID, add the Study ID to the Designs page (both as a filter and a column), move the Study ID on the Experiment page and the Study page, and add search functionality to each of the Experiment entities.

In short, he solved our problem but in return would like a number of new features, which I will create issues for. :)

**Comment by paul121** [comment]
Done - changed to a limit of 1500 items!
